packages/twenty-server/src/engine/core-modules/billing/services/billing-subscription.service.ts

```diff
@@ -0,0 +1,185 @@
+
+import {
+  BillingSubscription,
+  SubscriptionInterval,
+  SubscriptionStatus,
+} from 'src/engine/core-modules/billing/entities/billing-subscription.entity';
+import { StripeService } from 'src/engine/core-modules/billing/stripe/stripe.service';
+import { FeatureFlagKey } from 'src/engine/core-modules/feature-flag/enums/feature-flag-key.enum';
+import { FeatureFlagEntity } from 'src/engine/core-modules/feature-flag/feature-flag.entity';
+import { Workspace } from 'src/engine/core-modules/workspace/workspace.entity';
+import { EnvironmentService } from 'src/engine/integrations/environment/environment.service';
+
+@Injectable()
+export class BillingSubscriptionService {
+  protected readonly logger = new Logger(BillingSubscriptionService.name);
+  constructor(
+    private readonly stripeService: StripeService,
+    private readonly environmentService: EnvironmentService,
+    @InjectRepository(BillingSubscription, 'core')
+    private readonly billingSubscriptionRepository: Repository<BillingSubscription>,
+    @InjectRepository(FeatureFlagEntity, 'core')
+    private readonly featureFlagRepository: Repository<FeatureFlagEntity>,
+    @InjectRepository(Workspace, 'core')
+    private readonly workspaceRepository: Repository<Workspace>,
+  ) {}
+
+  /**
+   * @deprecated This is fully deprecated, it's only used in the migration script for 0.23
+   */
+  async getActiveSubscriptionWorkspaceIds() {
+    if (!this.environmentService.get('IS_BILLING_ENABLED')) {
+      return (await this.workspaceRepository.find({ select: ['id'] })).map(
+        (workspace) => workspace.id,
+      );
+    }
+
+    const activeSubscriptions = await this.billingSubscriptionRepository.find({
+      where: {
+        status: In([
+          SubscriptionStatus.Active,
+          SubscriptionStatus.Trialing,
+          SubscriptionStatus.PastDue,
+        ]),
+      },
+      select: ['workspaceId'],
+    });
+
+    const freeAccessFeatureFlags = await this.featureFlagRepository.find({
+      where: {
+        key: FeatureFlagKey.IsFreeAccessEnabled,
+        value: true,
+      },
+      select: ['workspaceId'],
+    });
+
+    const activeWorkspaceIdsBasedOnSubscriptions = activeSubscriptions.map(
+      (subscription) => subscription.workspaceId,
+    );
+
+    const activeWorkspaceIdsBasedOnFeatureFlags = freeAccessFeatureFlags.map(
+      (featureFlag) => featureFlag.workspaceId,
+    );
+
+    return Array.from(
+      new Set([
+        ...activeWorkspaceIdsBasedOnSubscriptions,
+        ...activeWorkspaceIdsBasedOnFeatureFlags,
+      ]),
+    );
+  }
+
+  async getCurrentBillingSubscription(criteria: {
+    workspaceId?: string;
+    stripeCustomerId?: string;
+  }) {
+    const notCanceledSubscriptions =
+      await this.billingSubscriptionRepository.find({
+        where: { ...criteria, status: Not(SubscriptionStatus.Canceled) },
+        relations: ['billingSubscriptionItems'],
+      });
+
+    assert(
```

Comment:
@charlesBochet I see, this one throws. Then I would call it getCurrentBillingSubscriptionOfFail to make it more obvious for getters?
To my comment above I guess stripeCustomerId will never be empty so up to you!